scripts/build.py

```diff
@@ -100,13 +115,95 @@ def copy_files_source_to_target(source, target):
                 continue
             if source not in source_path:
                 continue
-            target_path = source_path.replace(
-                source, target)
+
+            target_path = source_path
+            relative_filepath = os.path.relpath(source_path, source)
+            if (not relative_filepath.endswith(FILE_EXTENSIONS_NOT_TO_RENAME) or
+                    relative_filepath.endswith(FILE_ENDINGS_TO_RENAME)):
+                filepath, file_extension = os.path.splitext(source_path)
+                target_path = (filepath + '.' +
+                               file_hashes[relative_filepath]+
+                               file_extension)
+
+            target_path = target_path.replace(source, target)
             ensure_directory_exists(target_path)
             shutil.copyfile(source_path, target_path)
 
 
-def build_files(source, target):
+def generate_file_md5(filepath):
+    """Returns md5 hash of file.
+
+    Args:
+        filepath: str. Path to the file.
+
+    Returns:
+        str. Hexadecimal hash of specified file.
+    """
+    m = hashlib.md5()
+    with open(filepath, "rb") as f:
+        while True:
+            buf = f.read(HASH_BLOCK_SIZE)
+            if not buf:
+                break
+            m.update(buf)
+    return m.hexdigest()
+
+
+def get_file_hashes(folder_path):
+    """Returns hashes of all files in folder tree.
+
+    Args:
+        folder_path: str. Root folder of the tree.
+
+    Returns:
+        dict(str, str). Dictionary with keys specifying file paths and values
+        specifying file hashes.
+    """
+    file_hashes = dict()
+
+    print 'Processing %s' % os.path.join(os.getcwd(), folder_path)
```

Comment:
Can we be more specific: something like "Computing file hash for X"? Ditto below and elsewhere. This helps with debugging.